homeassistant/components/tado/tado_connector.py

```diff
@@ -174,13 +190,16 @@ def update_devices(self):
     def update_zones(self):
         """Update the zone data from Tado."""
         try:
-            zone_states = self.tado.get_zone_states()["zoneStates"]
+            if self.is_x:
+                zone_states = self.tado.get_zone_states()
+            else:
+                zone_states = self.tado.get_zone_states()["zoneStates"]
         except RuntimeError:
             _LOGGER.error("Unable to connect to Tado while updating zones")
```

Comment:
```suggestion
                zone_states = self.tado.get_zone_states()
        except RuntimeError:
            _LOGGER.error("Unable to connect to Tado while updating zones")
            
        if not self.is_x:
             zone_states = self.tado.get_zone_states()["zoneStates"]
```
just a bit less code

Reply:
I changed some of your changes since your suggested change would break the logic of the try...excpet block and had wrong indentation, but I think I got the gist of it and reduced the lines of code accordingly